zimbra/src/main/resources/public/ts/controllers/controller.ts

```diff
@@ -771,10 +771,40 @@ export let zimbraController = ng.controller("ZimbraController", [
             $scope.lightbox.show = true;
             template.open("lightbox", "move-mail");
         };
+        $scope.containsInternal = function(mails: string[]): boolean {
+            for (let mail of mails) {
+                if (!mail.includes("@")){
+                    return true;
+                }
+            }
+            return false;
+        }
+        $scope.containsExternal = function(mails: string[]): boolean {
+            for (let mail of mails) {
+                if (mail.includes("@")){
+                    return true;
+                }
+            }
+            return false;
+        }
+        $scope.getCurrentRecall = function(): Mail {
+            if ($scope.zimbra.currentFolder.mails.selection.selectedElements.length > 0)
+                return $scope.zimbra.currentFolder.mails.selection.selectedElements[0]
+            return $scope.mail;
+        }
+        $scope.isSelectedRecallable = function(): boolean {
+            return $scope.zimbra.currentFolder.mails.selection.selectedElements.length == 1 &&
+                $scope.isRecallable($scope.zimbra.currentFolder.mails.selection.selectedElements[0]);
+        }
+        $scope.isRecallable = function(mail: Mail): boolean {
+            return mail.returned == 'NONE' &&
+                $scope.containsInternal(<any>mail.to);
```

Comment:
`to` c'est pas un User[] ? (https://github.com/OPEN-ENT-NG/zimbra-connector/blob/dev/zimbra/src/main/resources/public/ts/model/mail.ts#L71)